### PR TITLE
Added Tracers architecture and sketched photo.py with the new classes

### LINKED ISSUE #16: Add Tracers class and test with GCph tracer
We would like to create a general abstract class `Tracer` that allows to pass information about the specific tracer used to compute kernels for the observables.

In particular, this task will implement:

- [x] Abstract Tracer class
- [x] First implementation of the Window function for GCph using the Tracer class

### PR DESCRIPTION
**Pull request by gcanasherrera**
In particular, this task will implement:

- [x] Abstract Tracer class
- [x] First skeleton for photo.py
- [x] First implementation of the Window function for GCph using the Tracer class
- [x] new data and jupyter notebook to interact with the class

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,79 @@
...
+- [Usage](#usage)
+- [Contributing](#contributing)
+- [License](#license)
+- [Acknowledgements](#acknowledgements)
+
+## Introduction
+Proof of concept of how to make CLOE (Cosmology Likelihood for Observables in Euclid) more efficient and user-friendly by transforming it into the Cosmology Library for Observables in Euclid (compatible with classic Boltzmann Solvers and JAX).
+This is a work in progress, and will benefit from the feedback of the Euclid community (and the whole cosmology community in general). 
+
+`cloelite` does not contain all the main features actual `cloe` has, and it should be seen as a toy repository to test further functionalities currently not implemented in `cloe` (and that would be painful to get with its current architecture).
```
Comment (marcobonici) [inline README.md]: I would consider to rephrase "would be painful to get" into "would be difficult to include".

File: cloelite/observables/tracer.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+"""
+## Author:
+    **Name**: G. Canas-Herrera & M. Bonici
+    **Date**: June 11, 2024
+
+## Notes:
+
+- Introducing the Tracer abstract class to implement different window functions
+- Idea behind: Tracer should be sufficiently generic to have flexibility (!)
+- SPIRIT: we do not classify among probes. All probes have their reason & place to be
```
Comment (marcobonici) [inline cloelite/observables/tracer.py]: I am not sure if this line should be around. I like this kind of thing for a presentation or discussion, not to be read in the documentation of a code.
No strong opinion, feel free to ignore.

File: cloelite/observables/tracer.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+- Lesson learnt: the cosmological analysis is a common endevaour
```
Comment (marcobonici) [inline cloelite/observables/tracer.py]: Same as above.

File: pyproject.toml [hunk 4]
```diff
@@ -7,9 +7,10 @@ license = "MIT"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.10"
+python = "^3.9"
 jax = "^0.4"
 jaxlib = "^0.4"
+euclidlib = "0.0.5"
```
Comment (marcobonici) [inline pyproject.toml:13]: Not yet.

File: cloelite/observables/photo.py [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+
+        Returns
+        -------
+        window: np.ndarray
+        """
+
+        pass
+
+class PositionsTracer(Tracer):
+    def __init__(self, perturbations: object, dndz: np.ndarray, z: np.ndarray,
```
Comment (marcobonici) [inline cloelite/observables/photo.py]: Perturbations should either be a `NonLinearPerturbations` object or an union set {`LinearPerturbations` , `NonLinearPerturbations` }.

File: cloelite/observables/photo.py [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+    **Date**: June 11, 2024
+
+## Notes:
+
+- Make sufficiently general to interface with CAMB keeping the structure by Cosmology
+
+"""
+
+class ShearTracer(Tracer):
+    def __init__(self, perturbations: object, dndz: np.ndarray, z: np.ndarray,
```
Comment (marcobonici) [inline cloelite/observables/photo.py]: Same as below.

File: cloelite/observables/photo.py [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+        r"""
+        A class to define the kernel for angular (galaxy) clustering
+
+        Initialize the cosmology class with given perturbations, redshift distribution, 
+        galaxy and magnification bias models, and nuisance parameters.
+
+        Parameters
+        ----------
+        perturbations : object
+            An object from NonLinearPerturbations class
```
Comment (marcobonici) [inline cloelite/observables/photo.py:145]: If the type accepted is an union set, update this docstring.

File: cloelite/observables/tracer.py (hunk 2 above)
Comment (gcanasherrera) [reply cloelite/observables/tracer.py]: Removed.

File: pyproject.toml (hunk 4 above)
Comment (gcanasherrera) [reply pyproject.toml:13]: Indeed, and never

File: cloelite/observables/tracer.py (hunk 3 above)
Comment (gcanasherrera) [reply cloelite/observables/tracer.py]: Removed.

File: cloelite/observables/photo.py (hunk 5 above)
Comment (gcanasherrera) [reply cloelite/observables/photo.py]: Ok!

File: cloelite/observables/photo.py (hunk 7 above)
Comment (gcanasherrera) [reply cloelite/observables/photo.py:145]: Done!

**Pull request merged**